### ISSUE TITLE
Regression: erased enum type not matched by an ArgumentFactory

### ISSUE THREAD

**Issue by martin-sladecek** [jdbi/jdbi#1646]
This used to work in 3.5.1, regression probably introduced in 3.6.x. (tested on 3.12.0)
```
public abstract class Entity<T> {
    private final T status;

    public T getStatus() { return status; }
}
```

cannot be used in SqlObject like this:
```
@SqlUpdate("INSERT INTO ENTITIES (STATUS) VALUES (:status)")
void insertEntity@BindBean Entity<?> entity);
```

The reason is that in 3.5.1, the actual value type was matched by against the ArgumentFactory, while now it's the actual generic type T that is being matched. The logic for this is in DelegatingArgumentFactory, which is not superclass of EnumArgumentFactory unfortunately.

**Comment by stevenschlansker** [comment]
hi @martin-sladecek , thanks for reporting this.  I have a fix at #1665 which will go out with `3.12.1`.  Please let us know if this fixes your problem 😸

**Comment by martin-sladecek** [comment]
The fix works, but unfortunately it cause another regression #1914
